### PR TITLE
feat(autogpt_builder): Monitor page v0.2

### LINKED ISSUE #7378: autogpt_builder: Monitor page v0.2
* Part 2 of implementing #7304
* Sequel of #7334
* See #7385 for further details

### PR DESCRIPTION
**Pull request by Pwuts**
### Background

* Sequel to #7335
* Part 2 of implementing #7304
* Resolves #7378

#### TODO
- [x] Choose a flow
- [ ] See all (status) information of the flow
  - [x] `Edit flow` button
- [ ] State introspection / debugging
  - [ ] Full "walkthrough" of current/last flow run, with all inputs/outputs inspectable
  - [ ] Highlight points of failure
- [x] List of recent runs
- [ ] List of flow versions
- [x] Stats on recent runs
  * duration
  * ~~cost~~ not currently implemented
  * chart?

### Changes 🏗️

- Improve responsive grid layout
  - Remove `container` class from `<main>` to utilize full screen width

- Improve detail views & add view for run details
  - Make flow run list entries selectable
  - Create `FlowInfo` and `FlowRunInfo` components
  - Improve layout of `FlowRunsStats`

- Improve ScrollableLegend spacing & styling
  - Hide scroll bar of scrollable legend
  - Center legend items if there is space left
  - Round icons
  - Vertically align icons with labels
  - FIX: Add condition to not display legend items for series with `legendType="none"`

- Add periodic 5s refresh of non-terminal flow run statuses
  - Split off `refreshFlowRuns(flowID)` from `fetchFlowsAndRuns()`
  - Add effect to run `refreshFlowRuns` every 5 seconds

- Improve and expand FlowRun info
  - Add `FlowRun.totalRunTime`: sum of the individual execution durations of all nodes
  - Add `FlowRun.endTime`
  - Use `NodeExecutionResult.add_time` instead of `start_time` as `FlowRun.startTime`

- Sort Flows by last executed

- Add icons to navbar items & hide unused items Backtrack and Explore

- Change UI mentions of "(Agent) Flow" to "Agent"

### Screenshots
image
image
image

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [x] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [x] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_builder/src/app/layout.tsx [hunk 1]
```diff
@@ -27,10 +28,18 @@ function ThemeProvider({ children, ...props }: ThemeProviderProps) {
...
-      <Link href="/build" className={buttonVariants({ variant: "ghost" })}>Build</Link>
-      <Link href="/backtrack" className={buttonVariants({ variant: "ghost" })}>Backtrack</Link>
-      <Link href="/explore" className={buttonVariants({ variant: "ghost" })}>Explore</Link>
+      <Link href="/monitor" className={buttonVariants({ variant: "ghost" })}>
+        <TimerIcon className="mr-1" /> Monitor
+      </Link>
+      <Link href="/build" className={buttonVariants({ variant: "ghost" })}>
+        <Pencil1Icon className="mr-1" /> Build
+      </Link>
+      {/* <Link href="/backtrack" className={buttonVariants({ variant: "ghost" })}>
```
Comment (aarushik93) [inline rnd/autogpt_builder/src/app/layout.tsx]: can we remove the commented out stuff altogether?

**Comment by aarushik93** [comment]
If you're renaming - should Edit Flow be Edit Agent too?

File: rnd/autogpt_builder/src/app/layout.tsx (hunk 1 above)
Comment (Pwuts) [reply rnd/autogpt_builder/src/app/layout.tsx]: sure

**Comment by Swiftyos** [comment]
Is the agent list scrollable in some way?

**Comment by aarushik93** [comment]
Is it possible to change the Agents list to show in most reccent order? Right now I see 5 days ago at the top and 2 minutes ago at the bottom

**Comment by Pwuts** [comment]
> Is the agent list scrollable in some way?
Not yet

**Comment by Pwuts** [comment]
> Is it possible to change the Agents list to show in most reccent order? Right now I see 5 days ago at the top and 2 minutes ago at the bottom

That is what this should do:
https://github.com/Significant-Gravitas/AutoGPT/pull/7385/files#diff-c0277d8d99528e90bb8cb9c6dccd0e76576618553ef642d5cd55c584e692973dR224-R229

Does it sort in the reverse order?

**Comment by aarushik93** [comment]
Mine was loaded from an old branch. looks good on this branch - thanks for checking 🙏🏽

**Review by aarushik93 — APPROVED** [review]

**Pull request merged**